eras/babbage/impl/src/Cardano/Ledger/Babbage/Rules/Utxow.hs

```diff
@@ -130,9 +130,29 @@ validateFailedBabbageScripts tx utxo = do
                   answer = one && (two || three)
                in answer
           )
-          (txscripts utxo tx)
-  failureUnless (Map.null failedScripts) $
-    Shelley.ScriptWitnessNotValidatingUTXOW (Map.keysSet failedScripts)
+          (txscripts utxo tx) -- txscripts = babbageTxScripts which includes reference scripts.
+   in failureUnless
+        (Map.null failedScripts)
+        (Shelley.ScriptWitnessNotValidatingUTXOW $ Map.keysSet failedScripts)
```

Comment:
is this still an accurate name? this test also validates reference scripts

Reply:
good point. I also think that the `Not` is weird in that name.